From a548cc319ae62b83a76ee5756d5f200f8c5cc86a Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Sun, 21 Apr 2013 13:01:03 +0300
Subject: ARM: i.MX6: cm-fx6: fix SPDIF output

Without this patch the spdif output shows no signal while playing back.
---
 arch/arm/mach-mx6/board-cm-fx6-q.h          |    2 +-
 arch/arm/plat-mxc/include/mach/iomux-mx6q.h |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-mx6/board-cm-fx6-q.h b/arch/arm/mach-mx6/board-cm-fx6-q.h
index abbe34e..d5e9403 100644
--- a/arch/arm/mach-mx6/board-cm-fx6-q.h
+++ b/arch/arm/mach-mx6/board-cm-fx6-q.h
@@ -157,7 +157,7 @@ static iomux_v3_cfg_t cm_fx6_q_common_pads[] = {
 static iomux_v3_cfg_t cm_fx6_q_spdif_pads[] = {
 	/* SPDIF */
 	MX6Q_PAD_GPIO_16__SPDIF_IN1,
-	MX6Q_PAD_GPIO_17__SPDIF_OUT1,
+	MX6Q_PAD_GPIO_19__SPDIF_OUT1,
 };
 
 static iomux_v3_cfg_t cm_fx6_q_can_pads[] = {
diff --git a/arch/arm/plat-mxc/include/mach/iomux-mx6q.h b/arch/arm/plat-mxc/include/mach/iomux-mx6q.h
index 557fc43..b56ea23 100644
--- a/arch/arm/plat-mxc/include/mach/iomux-mx6q.h
+++ b/arch/arm/plat-mxc/include/mach/iomux-mx6q.h
@@ -6106,7 +6106,7 @@
 #define  MX6Q_PAD_GPIO_19__ENET_1588_EVENT0_OUT		\
 		(_MX6Q_PAD_GPIO_19__ENET_1588_EVENT0_OUT | MUX_PAD_CTRL(NO_PAD_CTRL))
 #define  MX6Q_PAD_GPIO_19__SPDIF_OUT1		\
-		(_MX6Q_PAD_GPIO_19__SPDIF_OUT1 | MUX_PAD_CTRL(NO_PAD_CTRL))
+		(_MX6Q_PAD_GPIO_19__SPDIF_OUT1 | MUX_PAD_CTRL(MX6Q_SPDIF_OUT_PAD_CTRL))
 #define  MX6Q_PAD_GPIO_19__CCM_CLKO		\
 		(_MX6Q_PAD_GPIO_19__CCM_CLKO | MUX_PAD_CTRL(NO_PAD_CTRL))
 #define  MX6Q_PAD_GPIO_19__ECSPI1_RDY		\
-- 
1.7.9.5

